MaxText/checkpointing.py

```diff
@@ -58,11 +60,42 @@ def create_orbax_checkpoint_manager(
   return mngr
 
 
+def _find_idx(array: np.ndarray, replica_axis_idx: int):
+  """Returns the index along given dimension that the current host belongs to."""
+  idx = None
+  for idx, val in np.ndenumerate(array):
+    if val.process_index == jax.process_index():
+      break
+  return idx[replica_axis_idx]
+
+
+def _replica_devices(device_array: np.ndarray, replica_axis_idx: int):
+  """Returns the devices from the replica that current host belongs to.
+
+  Replicas are assumed to be restricted to the first axis.
+
+  Args:
+    device_array: devices of the mesh that can be obtained by mesh.devices()
+    replica_axis_idx: axis dimension along which replica is taken
+
+  Returns:
+    devices inside the replica that current host is in
+  """
+  idx = _find_idx(device_array, replica_axis_idx)
+  replica_result = np.take(device_array,
+                           idx,
+                           axis=replica_axis_idx)
+  return np.expand_dims(replica_result, axis=replica_axis_idx)
+
+
 def load_state_if_possible(checkpoint_manager: CheckpointManager,
                            data_iterator: Union[MultiHostDataLoadIterator, None],
                            load_parameters_from_path: str,
                            load_full_state_from_path: str,
                            abstract_unboxed_pre_state: train_state.TrainState,
+                           mesh,
+                           state_mesh_annotations,
+                           enable_single_replica_ckpt_restoring: bool,
                            dataset_type: Optional[str] = 'c4'):
   """Loads TrainState as possible from the inputs.
```

Comment:
if you're changing a signature, change the docstring.

Reply:
done, thanks.